### PR TITLE
Setup CI with GH Actions

### PR DESCRIPTION
**Pull request by marinalimeira**
[Jira: PATCHER-302](https://gruntwork.atlassian.net/browse/PATCHER-302)

- Test validation with `yarn test`
- Add check to the `dist/` folder, the versioned one should always be up-to-date.
- Configure ESLint and add it to the CI checks.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
```
Comment (marinalimeira) [inline .github/workflows/ci.yml:15]: This download is cached, in the logs it's possible to see the `Found in cache @ /opt/hostedtoolcache/node/18.17.1/x64` when installing `node`.

**Review by eak12913 — APPROVED** [review]
LGTM

**Comment by marinalimeira** [comment]
Thanks for reviewing!!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          node-version: '18'
+      - name: Install dependencies
+        run: yarn
+      - name: Run tests
```
Comment (oredavids) [inline .github/workflows/ci.yml:32]: This should be "Run lint"

**Pull request merged**